app/src/main/java/com/infomaniak/mail/ui/main/menu/MenuFoldersFragment.kt

```diff
@@ -53,17 +53,29 @@ abstract class MenuFoldersFragment : Fragment() {
     protected abstract val isInMenuDrawer: Boolean
 
     protected val defaultFoldersAdapter: FolderAdapter by lazy {
-        defaultFolderAdapter(isInMenuDrawer, onFolderClicked = ::onFolderSelected, onCollapseClicked = ::onFolderCollapse)
+        defaultFolderAdapter(
+            isInMenuDrawer,
+            onFolderClicked = ::onFolderSelected,
+            onCollapseClicked = ::onFolderCollapse,
+            onCollapseTransition = ::onCollapseTransition,
+        )
     }
 
     protected val customFoldersAdapter: FolderAdapter by lazy {
-        customFolderAdapter(isInMenuDrawer, onFolderClicked = ::onFolderSelected, onCollapseClicked = ::onFolderCollapse)
+        customFolderAdapter(
+            isInMenuDrawer,
+            onFolderClicked = ::onFolderSelected,
+            onCollapseClicked = ::onFolderCollapse,
+            onCollapseTransition = ::onCollapseTransition,
+        )
     }
 
     protected abstract fun onFolderSelected(folderId: String)
 
     protected abstract fun onFolderCollapse(folderId: String, shouldCollapse: Boolean)
 
+    protected abstract fun onCollapseTransition()
```

Comment:
Do we prefer to make it abstract so everyone needs to acknowledge it or do we simply implement it so it returns Unit by default and the children that want to override it take care of overriding it?

Reply:
Well I like to make it abstract because if you extends it you don't risk to forget about it
On the other hand this is not needed everywhere

I personnaly choose abstract as you can see, but I don't have a strong opinion on it